contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/HiveMetadataProvider.java

```diff
@@ -264,6 +265,10 @@ private HiveStats getStatsEstimateFromInputSplits(final List<LogicalInputSplit>
           final List<LogicalInputSplit> splits = Lists.newArrayList();
           final JobConf job = new JobConf(hiveConf);
           HiveUtilities.addConfToJob(job, properties);
+          if (AcidUtils.isTablePropertyTransactional(properties)) {
+            AcidUtils.setTransactionalTableScan(job, true);
+            HiveUtilities.setColumnTypes(job, properties, true, sd);
+          }
```

Comment:
It makes sense. Moreover since schema_evolution is required for acid tables [HIVE-12799](https://issues.apache.org/jira/browse/HIVE-12799) I've combined it with setColumnTypes() helper method.